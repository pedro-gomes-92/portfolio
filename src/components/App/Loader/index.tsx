import React, { FC } from 'react';
import { LoadingComponentProps } from 'react-loadable';

import Loader from 'components/Loader';
import { ReactComponent as LoaderSVG } from './assets/loader.svg';

const Component: FC<LoadingComponentProps> = ({ isLoading, ...rest }) => {
  return <Loader className="is-application" loader={<LoaderSVG />} isLoading={isLoading} {...rest} />;
};

export default Component;
