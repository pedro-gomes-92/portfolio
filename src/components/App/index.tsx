import React, { FC, useState } from 'react';
import { Container } from 'dots';

import Header from 'components/Header';
import Footer from 'components/Footer';
import Hero from 'components/Hero';

import About from 'sections/About';
import Projects from 'sections/Projects';
import Contact from 'sections/Contact';

const Component: FC = () => {
  const [offsetScroll, setOffsetScroll] = useState<number>(0);

  const [about, setAbout] = useState<HTMLElement>();
  const [projects, setProjects] = useState<HTMLElement>();
  const [contact, setContact] = useState<HTMLElement>();

  const elements = [about, projects, contact].filter(element => element !== undefined) as HTMLElement[];

  return (
    <>
      <Header
        reference={element => {
          const offset = element ? element.clientHeight : 0;
          setOffsetScroll(offset);
        }}
        elements={elements}
        offsetScroll={offsetScroll}
      />
      <Hero nextElement={about} offsetScroll={offsetScroll} />

      <Container>
        <About reference={element => setAbout(element)} />
        <Projects reference={element => setProjects(element)} />
        <Contact reference={element => setContact(element)} />
      </Container>
      <Footer />
    </>
  );
};

export default Component;
