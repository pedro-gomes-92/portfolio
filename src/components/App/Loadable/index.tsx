import Loadable from 'react-loadable';
import Loader from 'components/App/Loader';

const Component = Loadable({
  loader: () => import('components/App'),
  loading: Loader,
});

export default Component;
