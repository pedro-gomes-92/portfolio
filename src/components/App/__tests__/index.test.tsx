import React from 'react';
import { render } from '@testing-library/react';
import App from '../index';

describe('Given <App />', () => {
  describe('When initializes', () => {
    it('Then should match the snapshot', () => {
      const { container } = render(<App />);
      expect(container).toMatchSnapshot();
    });
  });
});
