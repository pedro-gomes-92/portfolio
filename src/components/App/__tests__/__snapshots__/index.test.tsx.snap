// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given <App /> When initializes Then should match the snapshot 1`] = `
<div>
  <div
    class="App"
  />
</div>
`;
